WooCommerce/WooCommerceTests/ViewRelated/Products/Edit Product/Subscription/SubscriptionTrialViewModelTests.swift

```diff
@@ -0,0 +1,161 @@
+import XCTest
+import Yosemite
+@testable import WooCommerce
+import WooFoundation
+
+final class SubscriptionTrialViewModelTests: XCTestCase {
+    private let samplePeriod: SubscriptionPeriod = .month
+
+    // MARK: `trialPeriodDescription`
+
+    func test_trialPeriodDescription_returns_expected_description_for_singular_trialPeriod_interval() {
+        // Given
+        let subscription = ProductSubscription.fake().copy(trialLength: "1", trialPeriod: samplePeriod)
+        let viewModel = SubscriptionTrialViewModel(subscription: subscription) { _, _, _ in }
+
+        // Then
+        XCTAssertEqual(viewModel.trialPeriodDescription, "month")
+    }
+
+    func test_trialPeriodDescription_returns_expected_description_for_plural_trialPeriod_interval() {
+        // Given
+        let subscription = ProductSubscription.fake().copy(trialLength: "2", trialPeriod: samplePeriod)
+        let viewModel = SubscriptionTrialViewModel(subscription: subscription) { _, _, _ in }
+
+        // Then
+        XCTAssertEqual(viewModel.trialPeriodDescription, "months")
+    }
+
+    // MARK: Completion block
+
+    func test_didTapDone_fires_completion_block_with_expected_values_when_no_changes_made() {
+        // Given
+        let subscription = ProductSubscription.fake().copy(trialLength: "1", trialPeriod: samplePeriod)
+
+        waitFor { promise in
+            let viewModel = SubscriptionTrialViewModel(subscription: subscription) { [weak self] trialLength, trialPeriod, hasUnsavedChanges in
+                guard let self else { return }
+                XCTAssertEqual(trialLength, "1")
+                XCTAssertEqual(trialPeriod, samplePeriod)
+                XCTAssertEqual(hasUnsavedChanges, false)
+                promise(())
```

Comment:
Nit: when testing async closures, assigning values to variables to test later (if possible) will make tests easier to read.

Reply:
Done in 6fe916627bd9157be53229acc842ede8286a602b